packages/provider/src/mobile.ts

```diff
@@ -10,6 +16,13 @@ initInpageProvider({
     commitSha: 'replace_commit_sha',
     version: 'replace_version',
   },
+  // WebView on Android doesn't support window.addEventListener('message')
+  addEventListener(listener) {
+    (document as any).addEventListener(LEATHER_WALLET_RPC_REQUEST, listener);
```

Comment:
```
declare global {
  interface DocumentEventMap {
    'leather-wallet-rpc-request': MessageEvent; 
  }
}
```
Should communicate clearer than casting entire `document` as any.  You'd also need to alter the type of the listener where it's defined:
```
addEventListener(listener: (this: Window | Document, ev: MessageEvent<any>) => any): void;
removeEventListener(listener: (this: Window | Document, ev: MessageEvent<any>) => any): void;
```

Reply:
Environment specific events concerns could be passed in using EventEmitter API. Provider packages doesn't need to know abut RN or web envs. Then no casting is needed.